eventlet/greenthread.py

```diff
@@ -170,6 +170,7 @@ def __init__(self, parent):
         greenlet.greenlet.__init__(self, self.main, parent)
         self._exit_event = event.Event()
         self._resolving_links = False
+        self._exit_funcs = None
```

Comment:
Why not unconditionally initialize `self._exit_funcs` to `deque()`?

Reply:
My only worry of wonder,
the problem with self._exit_funcs.remove((func, curried_args, curried_kwargs))
at any point the args or kwargs might be not the same order, and that might effect the match for value to remove

Reply:
Well, we'd be increasing resource usage for code clarity and fewer inline tests. But I take your point.
Another approach would be a breaking change: requiring the consuming application to explicitly designate that it intends to use `link()` and `unlink()`, and instantiating a `GreenThread` subclass containing `_exit_funcs` for such applications. But such a change is out of scope for this PR in any case, and may be completely untenable.